src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -2704,10 +2693,44 @@ namespace winrt::TerminalApp::implementation
             return nullptr;
         }
 
-        auto connection = existingConnection ? existingConnection : _CreateConnectionFromSettings(profile, controlSettings.DefaultSettings());
+        // Get the connection info for this set of settings.
+        auto connectionInfo = existingConnection ? nullptr :
+                                                   _CreateConnectionInfoFromSettings(profile, controlSettings.DefaultSettings());
+
+        // If we need to create a new connection, do that now, from the settings
+        // we just built.
+        auto connection = existingConnection ? existingConnection : _CreateConnectionFromInfo(connectionInfo);
+
+        // Finalize some defterm properties
         if (existingConnection)
         {
             connection.Resize(controlSettings.DefaultSettings().InitialRows(), controlSettings.DefaultSettings().InitialCols());
+
+            // If we had an existing connection (i.e., defterm), then get the
+            // current settings out of it and stash those with the control. It
+            // won't be perfect, but it'll be good enough.
+            if (auto conpty{ existingConnection.try_as<ConptyConnection>() })
+            {
+                connectionInfo = TerminalConnection::ConnectionInformation(winrt::name_of<TerminalConnection::ConptyConnection>(),
+                                                                           conpty.ToSettings());
```

Comment:
It feels like ConptyConnection (or, any connection) should be able to produce a fully-formed `ConnectionInformation` itself, without the consumer needing to know about its classname

Reply:
That... might? be able to be done. Not sure it's immediately relevant here (in the defterm context). But I guess it could be `ConptyConnection::CreateConnectionInfo(settings...) -> ConnectionInfo`, and have it fill in its own name.